File: Combinations.js
const getCombinations = (arr, n) => {
  let result = [];

  const combine = (temp, start) => {
    if (temp.length === n) {
      result.push([...temp]);
      return;
    }

    for (let i = start; i < arr.length; i++) {
      temp.push(arr[i]);
      combine(temp, i + 1);
      temp.pop();
    }
  };

  combine([], 0);
  return result;
};
console.log(
  getCombinations(
    [
      [1, 2],
      [3, 4],
      [5, 6],
      [7, 8],
      [9, 10],
    ],
    3
  )
);

File: combinations.py
def get_combi (arr,n):
  result = []

  def combi (element, idx):
    if (len(element) == n):
      result.append(element[:])
      return
    for i in range(idx, len(arr)):
      element.append(arr[i])
      combi(element, i+1)
      element.pop()

  combi([], 0)


  return result

print(get_combi([[1,2],[3,4],[5,6]], 2))